### DISCUSSION TITLE
Welcome to hyper-connections Discussions!

### DISCUSSION THREAD

**Discussion by lucidrains** [lucidrains/hyper-connections#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by markusheimerl** [comment]
So residual streams are a way for the model to "chose" how much to alter the current representation of the data by a addition of a transformation of the current representation. Do multiple residual streams give the opportunity to have multiple different alterations of the current representation come together in a sum? Im just wondering if I have the right mental picture here.

**Reply by lucidrains** [reply]
yes exactly, but it doesn't even have to be summed at the end, it could be attention pooled for instance

the design space is now thrown wide open given bytedance made it work

I think the entire field has overfit to this one residual stream, first order ODE. even interpretability has overfit to logit lens, but it could just be a consequence of the residual arch..

**Reply by lucidrains** [reply]
@markusheimerl anyways, i think experiments speak louder than words

you should give it a try and let us know what you see. I saw an immediate improvement in language modeling at 4 streams

**Comment by dame-cell** [comment]
Hey there ,So I'm still reading the paper and I noticed that in their paper they use LLMs for their experiments but training LLMs from scratch is not really feasible for everyone  I wanted to do some kind of small scale experiments  that will show results , 

now I do know that residual stream and and the newly proposed hyper-connections are mainly used in deep architectures but still would love any kind of suggestion 

Thanks in advance for any guidance!

**Reply by lucidrains** [reply]
yup, I have it integrated into x-transformers https://github.com/lucidrains/x-transformers

you can try the enwik8 example

**Reply by dame-cell** [reply]
Awesome as usual thank you so much for the help

**Comment by hyperevolnet** [comment]


**Reply by lucidrains** [reply]
besides the name, not really

**Comment by zhudefa** [comment]
Hello everyone, I am the author of Hyper-Connections. I would be more than happy to discuss anything related to Hyper-Connections with you. How have you been using them? How have they performed in your projects?

**Reply by lucidrains** [reply]
@zhudefa greatly enjoyed your paper! 🙏
